### PR TITLE
fix(postgres): postgres setup

### PR DESCRIPTION
**Pull request by jensenity**
This PR fixes the error in the init.sql. Postgres SQL doesn't support `CREATE DATABASE IF NOT EXISTS`, so I have added a psql shell script in the init.sh to check whether there's the database we want, if not `create database datahub`. And also some minor changes in the sql file. Such as we can use the TEMP table syntax when we create a temp table.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: docker/postgres-setup/Dockerfile [hunk 1]
```diff
@@ -6,6 +6,6 @@ COPY docker/postgres-setup/init.sql /init.sql
 COPY docker/postgres-setup/init.sh /init.sh
 RUN chmod 755 init.sh
 
-ENV DATAHUB_DB_NAME="datahub"
+ENV DATAHUB_DB_NAME=$POSTGRES_DATABASENAME
```
Comment (dexter-mh-lee) [inline docker/postgres-setup/Dockerfile:9]: Why do we need a separate env variable?

Comment (dexter-mh-lee) [reply docker/postgres-setup/Dockerfile:9]: Can we just use DATAHUB_DB_NAME ?

Comment (jensenity) [reply docker/postgres-setup/Dockerfile:9]: I guess we don't need it. Initial intention is for the users to be able set their own database name through the env. But if it's unnecessary, we can remove it!

Comment (dexter-mh-lee) [reply docker/postgres-setup/Dockerfile:9]: So DATAHUB_DB_NAME already does that. Here it seems like it's always set to "datahub" but it does use the appropriate db name during setup.(This is the case for mysql-setup, where I tested with different db names)

Comment (jensenity) [reply docker/postgres-setup/Dockerfile]: true...dbeec80

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**